File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/ShippingAddressAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces;

import com.emerchantpay.gateway.genesisandroid.api.models.Country;
import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface ShippingAddressAttributes {

    RequestBuilder requestBuilder = new RequestBuilder("");

    // Shipping PaymentAddress
    default ShippingAddressAttributes setShippingFirstname(String firstname) {
        requestBuilder.addElement("first_name", firstname);
        return this;
    }

    default ShippingAddressAttributes setShippingLastname(String lastname) {
        requestBuilder.addElement("last_name", lastname);
        return this;
    }

    default ShippingAddressAttributes setShippingPrimaryAddress(String primaryAddress) {
        requestBuilder.addElement("address1", primaryAddress);
        return this;
    }

    default ShippingAddressAttributes setShippingSecondaryAddress(String secondaryAddress) {
        requestBuilder.addElement("address2", secondaryAddress);
        return this;
    }

    default ShippingAddressAttributes setShippingCity(String city) {
        requestBuilder.addElement("city", city);
        return this;
    }

    default ShippingAddressAttributes setShippingZipCode(String zipCode) {
        requestBuilder.addElement("zip_code", zipCode);
        return this;
    }

    default ShippingAddressAttributes setShippingState(String state) {
        requestBuilder.addElement("state", state);
        return this;
    }

    default ShippingAddressAttributes setShippingCountry(String country) throws IllegalAccessException {
        Country c = new Country();

        requestBuilder.addElement("country", c.getIsoCode(country));

        return this;
    }

    default RequestBuilder buildShippingAddress() {
        return requestBuilder;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/ShippingAddressAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces

import com.emerchantpay.gateway.genesisandroid.api.models.Country
import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder



interface ShippingAddressAttributes {

    // Shipping PaymentAddress
    fun setShippingFirstname(firstname: String): ShippingAddressAttributes {
        requestBuilder.addElement("first_name", firstname)
        return this
    }

    fun setShippingLastname(lastname: String): ShippingAddressAttributes {
        requestBuilder.addElement("last_name", lastname)
        return this
    }

    fun setShippingPrimaryAddress(primaryAddress: String): ShippingAddressAttributes {
        requestBuilder.addElement("address1", primaryAddress)
        return this
    }

    fun setShippingSecondaryAddress(secondaryAddress: String): ShippingAddressAttributes {
        requestBuilder.addElement("address2", secondaryAddress)
        return this
    }

    fun setShippingCity(city: String): ShippingAddressAttributes {
        requestBuilder.addElement("city", city)
        return this
    }

    fun setShippingZipCode(zipCode: String): ShippingAddressAttributes {
        requestBuilder.addElement("zip_code", zipCode)
        return this
    }

    fun setShippingState(state: String): ShippingAddressAttributes {
        requestBuilder.addElement("state", state)
        return this
    }

    @Throws(IllegalAccessException::class)
    fun setShippingCountry(country: String): ShippingAddressAttributes {
        val c = Country()

        requestBuilder.addElement("country", c.getIsoCode(country)!!)

        return this
    }

    fun buildShippingAddress(): RequestBuilder {
        return requestBuilder
    }

    companion object {

        val requestBuilder = RequestBuilder("")
    }
}